api/src/main/java/io/onetable/model/storage/OneDataFile.java

```diff
@@ -46,13 +44,12 @@ public class OneDataFile {
   // file format
   @Builder.Default @NonNull FileFormat fileFormat = FileFormat.APACHE_PARQUET;
   // partition ranges for the data file
-  @Builder.Default @NonNull Map<OnePartitionField, Range> partitionValues = Collections.emptyMap();
+  @Builder.Default @NonNull List<PartitionValue> partitionValues = Collections.emptyList();
```

Comment:
This and the below usage of list are the main changes

Reply:
How much a difference would it be in usage of map vs list ? is it significant ? and if so is it due to continous location of data ?

Reply:
I found this breakdown here for the memory differences that is pretty informative https://developer.ibm.com/articles/j-codetoheap/ - There is a 36 bytes per entry overhead that exists for your standard HashMap implementation. There is an 8 byte overhead per element in the array list implementation. Some of these gains are lost by adding a new wrapper object for the `PartitionValue`. 

We'll get a larger win in the case of the column stats where we're just adding a field to the existing object that was in the map. It will remove 24 bytes per field tracked per file. So if we have thousands of files with a large schema, this will add up.

Another aspect is that we don't need a keyed lookup anywhere and it is easier to allocate a list with the appropriate size than it is to get a Map with the right size considering the load-factor for maps.

Reply:
Thanks for the detailed info.